### ISSUE TITLE
Files with spaces can't be accessed by the default file index

### ISSUE THREAD

**Issue by svenstaro** [actix/actix-web#653]
Sample code:

```rust
use actix_web::{fs, server, App};

fn main() {
    let sys = actix::System::new("static_index");

    server::new(|| {
        App::new().handler("/", fs::StaticFiles::new(".").unwrap().show_files_listing())
    })
    .bind("127.0.0.1:8080")
    .expect("Can not start server on given IP/Port")
    .start();

    println!("Started http server: 127.0.0.1:8080");
    let _ = sys.run();
}
```

run this in a directory with a file called `hello world.txt`. Try to access the file. The generated URL will be `127.0.0.01:8080/hello%20world.txt` (which is correct) but then the route will give back a 404.

The default file index should definitely just work with files with spaces in them.

This was originally reported as https://github.com/svenstaro/miniserve/issues/23.

**Comment by DoumanAsh** [comment]
Fixed in https://github.com/actix/actix-web/commit/4d45313f9d21c5b9835f5effc260c54a4507ce84

**Comment by svenstaro** [comment]
Cool stuff! Could you make a release with this fix?

**Comment by DoumanAsh** [comment]
Ok, I'll do it later on and let you know

**Comment by DoumanAsh** [comment]
@svenstaro I published `0.7.18`

**Comment by svenstaro** [comment]
Sweet!
